lib/AST/Decl.cpp

```diff
@@ -4220,17 +4220,18 @@ bool ValueDecl::isUsableFromInline() const {
 
 bool ValueDecl::isInterfacePackageEffectivelyPublic() const {
   // If a package decl has a @usableFromInline (or other inlinable)
-  // attribute, and is defined in a module built from interface, it
-  // can be referenced by another module that imports it even though
-  // the defining interface module does not have package-name (such
-  // as public or private interface); in such case, the decl is treated
-  // as public and access checks in sema are skipped.
+  // attribute, and is defined in an interface module (or built from
+  // interface), it can be referenced by another module that imports
+  // it even though the defining interface module does not have
+  // package-name (public/private interface); in such case, the decl
+  // is treated as public and access checks in sema are skipped.
   // We might need to add another check here to ensure the interface
   // was part of the same package before the package-name was removed.
   return getFormalAccess() == AccessLevel::Package &&
          isUsableFromInline() &&
          getModuleContext()->getPackageName().empty() &&
-         getModuleContext()->isBuiltFromInterface();
+         (getModuleContext()->isBuiltFromInterface() ||
+          getDeclContext()->getParentSourceFile()->Kind == SourceFileKind::Interface);
```

Comment:
Sometimes there isn't a source file for a decl, so the code should ideally have a defensive `nullptr` check for the result of `getParentSourceFile()`.

Reply:
Good catch

Reply:
updated